streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java

```diff
@@ -225,9 +218,13 @@ public boolean equals(final Object o) {
             Objects.equals(headers, that.headers);
     }
 
+    /**
+     * Equality is implemented in support of tests, *not* for use in Hash collections, since this class is mutable.
+     */
+    @Deprecated
```

Comment:
No need to deprecate internal method. If we want to remove it, we should just do it.

Reply:
We can't remove `hashCode`, since it's defined by `Object`. Marking it as deprecated isn't necessary, since any attempt to use it results in an exception, but it's just extra documentation that `hashCode` is not to be used on `ProcessorRecordContext`.

Reply:
What I meant by "remove" was, to remove the overwrite :)

Thanks to clarify why you want to overwrite and throw. But this does not mean that the method is deprecated... Using the annotation implies that we want to remove this overwrite in the future was is miss leading. I would prefer to remove the annotation. \cc @guozhangwang @bbejeck

Reply:
I think that throwing the exception in its override implementation is sufficient for future bug-proof, as regarding to documentation I feel neutral either way.

Reply:
We should still add JavaDocs and explain that we throw and why. But deprecation != documentation IMHO.